bob-backend/src/pearl/group.rs

```diff
@@ -233,6 +239,34 @@ impl Group {
         }
     }
 
+    async fn safe_timestamp_step(&self) -> Option<u64> {
+        if let Some(adjust) = self.settings.config().skip_holders_by_timestamp_step_when_reading_sec() {
+            let mut start_timestamps = self.holders.read().await
+                .iter()
+                .map(|holder| holder.start_timestamp())
+                .collect::<Vec<u64>>();
+            start_timestamps.sort();
+            start_timestamps.push(get_current_timestamp());
+            let mut max_diff = None;
+            for i in 0..start_timestamps.len() - 1 {
+                let diff = start_timestamps[i + 1] - start_timestamps[i];
+                if let Some(max_diff_v) = max_diff {
+                    if max_diff_v < diff {
+                        max_diff = Some(diff);
+                    }
+                } else {
+                    max_diff = Some(diff);
+                }
+            }
+            if let Some(max_diff) = max_diff {
```

Comment:
If `max_diff` is None, then init it to `self.settings.timestamp_period_as_secs()`:
```
let max_diff = max_diff.unwrap_or(self.settings.timestamp_period_as_secs());
```